### ISSUE TITLE
Server crashes immediately when player enters Twilight Forest portal with spongeforge

### ISSUE THREAD

**Issue by LuminescentMoon** [PG85/OpenTerrainGenerator#141]
Using

- Minecraft 1.12.2
- Forge 2611
- SpongeForge 2953 (tested with multiple past versions as well)
- OTG v4 & v5
- Twilight Forest 3.6.345

[Crash Log](https://hastebin.com/obaniqujic.vbs)
[Full Server Mod List](https://hastebin.com/yimepasuju.vbs) (slightly old)

Server crashes immediately when a player enters a Twilight Forest portal. Relaunching the server and attempting to rejoin with the same player character will cause the server to crash again, requiring a world backup to remedy the issue.

**Removing either SpongeForge or OTG fixes the issue.**

This issue is not due to FoamFix. This is tested by attempting to reproduce the crash with FoamFix removed with success.

This issue does **NOT** occur when entering a portal from [The Betweenlands](https://minecraft.curseforge.com/projects/angry-pixel-the-betweenlands-mod), a mod similar to Twilight Forest in that it adds a new world that has its own separate content.

**Comment by LuminescentMoon** [comment]
Most likely related to #122

**Comment by LuminescentMoon** [comment]
Still not fixed in v6

**Comment by PG85** [comment]
Hey, could you try v7/v8 from the discord at #openterraingen and see if that fixes the issue? Feel free to reopen this issue if the problem persists.

**Comment by bluelightning32** [comment]
I tried using OpenTerrainGenerator-1.12.2_-_v8_r1.jar from discord, but then the clients from the modpack I'm running can't connect. Also I had to do a world reset to get it to start with v8 because of a biome id mismatch.

When will a new version of OTG be released to curseforge so that I can ask the modpack author to include the new version?

**Comment by bluelightning32** [comment]
I'm running:
Modpack: Roguelike Adventures and Dungeons - v1.29
Forge: forge-1.12.2-14.23.5.2836-universal.jar
SpongeForge: spongeforge-1.12.2-2838-7.1.7-RC3882.jar
OTG: OpenTerrainGenerator-1.12.2 - v6.jar
TF: twilightforest-1.12.2-3.9.984-universal.jar

Initially I started the server with just Forge. I later converted it to SpongeForge without resetting the world. Entering the twilight forest would crash the server as described in this bug.

I worked around this by using NBTExplorer to set the generatorOptions field to an empty string inside of Biome Bundle/DIM7/level.dat. It was set to the string "OpenTerrainGenerator".
